admin/digilan-admin.php

```diff
@@ -395,6 +402,128 @@ private static function updateCityscopeCloud($cityscopeCloud)
+                continue;
+            }
+
+            if ($lang) {
+                $new_field_data[$key][$lang] = $value;
+            } else {
+                $new_field_data[$key] = $value;
+            }
+        }
+
+        // Remove unused property
+        $new_field_data_filtered = array_filter($new_field_data, function($data) {
+            return is_array($data) ? array_filter($data) : $data;
+        });
+
+        if ($new_field_data_filtered['type'] === 'tel') {
+            $new_field_data_filtered['regex'] = '^\+?(?:[0-9]\s?){6,14}[0-9]$';
+        } elseif ($new_field_data_filtered['type']  == 'email') {
+            $new_field_data_filtered['regex'] = '^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$';
+        }
+
+        $first_translation_name = current($new_field_data_filtered['display-name']);
+        $new_field_key = str_replace(' ', '-', strtolower($first_translation_name));
+        if ($fields[$new_field_key]) {
+            \DLT\Notices::addError(__('Field name already exist', 'digilan-token'));
+            wp_redirect(self::getAdminUrl('form-settings'));
+            exit();
+        }
+
+        $positions = array_map('self::get_positions', $fields);
+        $new_pos = max($positions) + 1;
+        $new_field_data_filtered['position'] = $new_pos;
+
+        $fields[$new_field_key] = array_filter($new_field_data_filtered);
+        update_option('user_form_fields', $fields);
+        \DLT\Notices::addSuccess(__('New field added', 'digilan-token'));
+        wp_redirect(self::getAdminUrl('form-settings'));
+        exit();
+    }
+
+    private static function update_user_form_fields()
+    {
+        $user_form_fields = get_option('user_form_fields');
+        $POST_form_key = array_keys($_POST);
+        $deleted_keys = array();
+
+        foreach($POST_form_key as $form_key) {
+            // form_key example : "form-fields/display-name/first_name/fr_FR"
+            [$prefix, $field_option, $field_key, $lang_code] = explode('/', $form_key);
+            if ($prefix !== 'form-fields') {
+                continue;
+            }
+            if (in_array($field_key, $deleted_keys)) {
+                continue;
+            }
+            if ($field_option == 'delete') {
+                array_push($deleted_keys, $field_key);
+                unset($user_form_fields[$field_key]);
+                continue;
+            }
+
+            $input_type = $user_form_fields[$field_key]['type'];
+            $is_input_radio_or_select = in_array($input_type, array('radio', 'select'));
+            $is_option = ($field_option === 'options');
+
+            $value;
+            if ($is_input_radio_or_select && $is_option) {
+                $opts_exploded = explode(',', $_POST[$form_key]);
+                $opts_trimmed = array_map('trim', $opts_exploded);
+                $value = $opts_trimmed;
+            } else {
+                $value = $_POST[$form_key];
```

Comment:
no sanitize seen , am i missing it ? => should be more visible , is it not here? => must be done

Reply:
Sanitize has been added